File: src/Q1057_Tournament.java
import java.util.Scanner;
 
public class Q1057_Tournament {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        scan.nextInt();
        int a = scan.nextInt();
        int b = scan.nextInt();
 
        int count = 0;
        while (a != b) {
            count++;
            a = (int) Math.ceil(a / 2.0);
            b = (int) Math.ceil(b / 2.0);
        }
 
        System.out.println(count);
        scan.close();
    }
}

File: src/scala/Q1057_Tournament.scala
package scala

object Q1057_Tournament {
  def main(args : Array[String]) : Unit = {
    var Array(n, a, b) = readLine.split(" ").map(_.toInt)
    var count = 0
    while (a != b) {
      a = next(a)
      b = next(b)
      count += 1
    }
    println(count)
  }

  def next(x: Double): Int = Math.ceil(x / 2).toInt
}
